src/nomad_simulations/schema_packages/properties/spectral_profile.py

```diff
@@ -101,12 +101,45 @@ def resolve_pdos_name(self, logger: 'BoundLogger') -> Optional[str]:
             )
             return None
 
-        # Resolve the `name` from the `entity_ref`
         name = None
+        # If the reference is to an AtomsState, use its atom_definition_ref
```

Comment:
While I'm fine with the logic at hand, it's kinda clumsy having all this logic in one place.
I can moreover see other sections using similar functionalities.
I think that this could be better handled with (a few) abstraction layer(s) that define the necessary behavior obtain representations.